### PR TITLE
Fix potential issue from external cfg data

### PR DESCRIPTION
**Pull request by gdong1**
Currently SpiLoadExternalConfigData will update the output buffer even
signature or other checks failure. It may cause issue to caller when
caller consume the data.
This patch update the logic to update output buffer only load external
config data success.

Signed-off-by: Guo Dong <guo.dong@intel.com>

### CODE REVIEW

File: Platform/CommonBoardPkg/Library/BoardSupportLib/BoardSupportLib.c [hunk 1]
```diff
@@ -164,11 +163,11 @@ SpiLoadExternalConfigData (
   //
   Status = EFI_UNSUPPORTED;
   if (CfgDataLoadSrc == FlashRegionPlatformData) {
-    Status = SpiFlashRead (FlashRegionPlatformData, Address, BlobSize, Buffer);
+    Status = SpiFlashRead (FlashRegionPlatformData, Address, BlobSize, (UINT8*)&CfgBlob);
```
Comment (mauricema) [inline Platform/CommonBoardPkg/Library/BoardSupportLib/BoardSupportLib.c]: Add a space between  UINT and *  for  UINT8*

File: Platform/CommonBoardPkg/Library/BoardSupportLib/BoardSupportLib.c [hunk 2]
```diff
@@ -141,7 +141,7 @@ SpiLoadExternalConfigData (
   UINT32       Address;
   UINT32       BlobSize;
   UINT8       *Buffer;
-  CDATA_BLOB  *CfgBlob;
+  CDATA_BLOB  CfgBlob;
```
Comment (mauricema) [inline Platform/CommonBoardPkg/Library/BoardSupportLib/BoardSupportLib.c]: align the variable names ?

**Comment by mauricema** [comment]
What was the real issue again here?   
I think the output buffer can be updated in any case (either fail or success).   Caller needs to rely on 'Status' returned to determine if the data is good to consume.   If Status indicates failure, even if the data buffer is filled,  it should be ignored by the caller.

**Comment by gdong1** [comment]
After double check the caller, looks the caller doesn't use these data if status is not success.
so this patch is not necessary.

**Pull request closed without merging**